File: solve-boolean.cpp
```c++
bool solveBoolean(std::string expression) {
    bool result = true;
    for(int i=0; i<expression.length(); i++){
        if(expression[i] == '&') {
            result &= true;
        } else if(expression[i] == '|') {
            result |= true;
        } else{
            if(expression[i] == 'T')
                result = true;
            else
                result = false;
        }
    }
    return result;
}

File: solve-boolean.py
def solve_boolean(input_str):
    if input_str == "T":
        return True
    elif input_str == "F":
        return False
    elif "&" in input_str and "|" in input_str:
        raise ValueError("Invalid expression")
    else:
        result = eval("True" + input_str.replace("&", "and").replace("|", "or"))
        return result